File: templates/hytale/optional/java/src/main/java/{{PACKAGE_PATH}}/ServerConfig.java
package {{PACKAGE_NAME}};

import com.hypixel.hytale.codec.Codec;
import com.hypixel.hytale.codec.KeyedCodec;
import com.hypixel.hytale.codec.builder.BuilderCodec;

public class ServerConfig {

    public static final BuilderCodec<ServerConfig> CODEC =
            BuilderCodec.builder(ServerConfig.class, ServerConfig::new)

                    .append(
                            new KeyedCodec<String>("ModName", Codec.STRING),
                            (config, value, info) -> config.modName = value,
                            (config, info) -> config.modName
                    )
                    .add()

                    .build();

    private String modName = "{{PROJECT_NAME}}";

    public ServerConfig() {}

    public String getModName() {
        return modName;
    }

    public void setModName(String modName) {
        this.modName = modName;
    }
}

File: templates/hytale/optional/kotlin/src/main/kotlin/{{PACKAGE_PATH}}/ServerConfig.kt
package {{PACKAGE_NAME}}

import com.hypixel.hytale.codec.Codec
import com.hypixel.hytale.codec.KeyedCodec
import com.hypixel.hytale.codec.builder.BuilderCodec

class ServerConfig {

    companion object {

        @JvmField
        val CODEC: BuilderCodec<ServerConfig> =
            BuilderCodec.builder(ServerConfig::class.java, ::ServerConfig)

                .append(
                    KeyedCodec<String>("ModName", Codec.STRING),
                    { config, value, _ -> config.modName = value },
                    { config, _ -> config.modName }
                )
                .add()

                .build()
    }

    var modName: String = "{{PROJECT_NAME}}"
}